src/datamodel/builder/Uuid.cpp

```diff
@@ -10,3 +10,99 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/builder/Uuid.h"
+
+#include <random>
+#include <stdexcept>
+
+namespace uprotocol::datamodel::builder {
+
+struct UuidBuilder::UuidSharedState {
+	std::mt19937_64 random_engine{std::random_device{}()};
+	uint16_t counter{0};
+	uint64_t last_unix_ts_ms{0};
+	uint64_t rand_b{0};
+	bool rand_b_initialized{false};
```

Comment:
These values may need to be wrapped in `std::atomic` since this state will be shared across UuidBuilder instances

Reply:
Given the UUIDv7 change is coming, we can hold off on messing around with the shared state stuff. It can remain unshared for this PR.